src/typing/typecheck.ml

```diff
@@ -415,6 +416,11 @@ let rec infer_comp ({Location.it=c; at} : Desugared.comp) : (Syntax.comp * Mlty.
      check_comp c Mlty.Judgement >>= fun c ->
      return (locate ~at (Syntax.Fresh (xopt, c)), Mlty.Judgement)
 
+   | Desugared.AbstractAtom (c1, c2) ->
```

Comment:
Since in the operational semantics you allow `c2` to be a judgement or a boundary, typechecking should also allow `c2` to be a judgement or a boundary. We have `Tyenv.as_judgement_or_boundary` for that purpose, see how it's used elsewhere in this file. You infer the type of `c2` and then pump it through `Tyenv.as_judgement_or_boundary`. The final type depends on the type of `c2`.